### PR TITLE
Refactoring Grafana chart

### PR DESCRIPTION
**Pull request by willvrny**
# Description

In these change I've pulled most of the configuration out into files or templates to prevent the values file getting too large. When running helm with '--dry-run' and '--debug' options it makes it harder to check the output. I've also added some dashboards and managed to replace the job with watcher configuration.

# Changes

- Moved dashboards from the values files to a separate folder in the chart.
- Moved the datasource configuration to a separate folder in the chart.
- Configured the watcher to set the datasource and removed the now redundant setDatasource job.
- Added some default dashboards for monitoring nodes, deployments, pods and resource requests.
- Moved the grafana.ini file to a template.
- Fixed the version of Grafana instead of using 'latest'.

### CODE REVIEW

**Comment by swade1987** [comment]
@willvrny I really like the extendibility of the dashboards and data sources now. I would suggest we add an `README.md` file to explain how new dashboards and data sources should be added to the chart.

**Comment by willvrny** [comment]
@swade1987 I agree. We might want to explain the reasoning behind the dashboards as well?

File: charts/grafana/templates/deployment.yaml [hunk 1]
```diff
@@ -58,6 +58,8 @@ spec:
           volumeMounts:
             - name: config-volume
               mountPath: {{ default "/etc/grafana" .Values.server.configLocalPath | quote }}
+            - name: dashboard-volume
```
Comment (dkoshkin) [inline charts/grafana/templates/deployment.yaml]: If I understand this PR correctly this is now using a directory watcher to load dahsboards/datasources into Grafana. 

This feature is buggy and being deprecated and reworked by the Grafana team. When loading dashboards from a path they behave differently than other dashboards. 
* You probably noticed that they are repeated in the dropdown list
* You can't star them and they dont show up under recents
* It is not possible to setup alerts for these dashboards.

This was the reason to use a sidecar container to load the dashboards so they behave as any other chart.

Comment (swade1987) [inline charts/grafana/templates/deployment.yaml]: The PR removes does two things:

1. Removes the need for the datasource job
2. Makes the dashboard config map easier to manage and extend

Comment (willvrny) [inline charts/grafana/templates/deployment.yaml]: @dkoshkin @swade1987 the original chart was already loading the dashboards in via the watcher. This change does remove the job and use the watcher to load the datasource.

**Comment by dkoshkin** [comment]
Closing for now, lets work on this with the updated branch

**Pull request closed without merging**